src/js/components/leftPanel/layersPanel/GenericLayerControl.tsx

```diff
@@ -27,6 +30,12 @@ const GenericLayerControl = (props: LayerControlProps): React.ReactElement => {
   //Determine if we need density control on this layer
   const densityPicker = layer && densityEnabledLayers.includes(layer.id);
 
+  useEffect(() => {
+    if (layer && (layer as any)?.visible === false) {
+      mapController.resetDefinedDateRange(layer.id);
+    }
+  }, [layer?.visible]);
```

Comment:
I do not think we need this useEffect that fires for any layer toggle action. We should have fires visibility being handled in mapController.toggleLayerVisibility or in the visibility component itself